### PR TITLE
Create InvalidCheckpoint Error to bucket errors relevant to checkpoint

### PR DESCRIPTION
**Pull request by Sevenannn**
## What changes are proposed in this pull request?
Create a new variation of error: InvalidCheckpoint error to bucket errors relevant to invalid checkpoint files. This will allow for a more flexible handling of checkpoint related errors when using delta kernel crate.

## How was this change tested?
Via `cargo t --all-features --all-targets`

### CODE REVIEW

**Review by scovich — APPROVED** [review]
Hi! Thanks for proposing this change, it looks great.

**Review by roeap — APPROVED** [review]

File: kernel/src/log_segment.rs [hunk 1]
```diff
@@ -68,7 +68,7 @@ impl LogSegment {
         {
             require!(
                 checkpoint_file.version + 1 == commit_file.version,
-                Error::generic(format!(
+                Error::invalid_checkpoint(format!(
```
Comment (zachschuermann) [inline kernel/src/log_segment.rs:71]: nit: format! is already a String so we can avoid the clone
```suggestion
                Error::InvalidCheckpoint(format!(
```

File: kernel/src/log_segment.rs [hunk 2]
```diff
@@ -369,7 +369,7 @@ fn list_log_files_with_checkpoint(
             latest_checkpoint.version
         );
     } else if checkpoint_parts.len() != checkpoint_metadata.parts.unwrap_or(1) {
-        return Err(Error::Generic(format!(
+        return Err(Error::invalid_checkpoint(format!(
```
Comment (zachschuermann) [inline kernel/src/log_segment.rs:372]: same as above
```suggestion
        return Err(Error::InvalidCheckpoint(format!(
```

**Review by zachschuermann — APPROVED** [review]
thanks this is a good idea - just a small nit but otherwise LGTM!

File: kernel/src/log_segment.rs (hunk 1 above)
Comment (Sevenannn) [reply kernel/src/log_segment.rs:71]: Thanks! That makes sense, let me update this

**Comment by nicklan** [comment]
Tested msrv locally, not sure why the test is failing

**Comment by nicklan** [comment]
Thanks for the contribution!

**Pull request merged**